token/client/src/token.rs

```diff
@@ -2381,143 +2348,81 @@ where
         .await
     }
 
-    /// Create an equality proof context state account for a confidential
-    /// transfer.
-    #[allow(clippy::too_many_arguments)]
-    pub async fn create_equality_proof_context_state_for_transfer<S: Signer>(
-        &self,
-        context_state_account: &Pubkey,
-        context_state_authority: &Pubkey,
-        equality_proof_data: &CiphertextCommitmentEqualityProofData,
-        equality_proof_signer: &S,
-    ) -> TokenResult<T::Output> {
-        // create equality proof context state
-        let instruction_type = ProofInstruction::VerifyCiphertextCommitmentEquality;
-        let space = size_of::<ProofContextState<CiphertextCommitmentEqualityProofContext>>();
-        let rent = self
-            .client
-            .get_minimum_balance_for_rent_exemption(space)
-            .await
-            .map_err(TokenError::Client)?;
-
-        let equality_proof_context_state_info = ContextStateInfo {
-            context_state_account,
-            context_state_authority,
-        };
-
-        self.process_ixs(
-            &[
-                system_instruction::create_account(
-                    &self.payer.pubkey(),
-                    context_state_account,
-                    rent,
-                    space as u64,
-                    &zk_token_proof_program::id(),
-                ),
-                instruction_type.encode_verify_proof(
-                    Some(equality_proof_context_state_info),
-                    equality_proof_data,
-                ),
-            ],
-            &[equality_proof_signer],
-        )
-        .await
-    }
-
-    /// Create a ciphertext validity proof context state account for a
-    /// confidential transfer.
-    pub async fn create_ciphertext_validity_proof_context_state_for_transfer<S: Signer>(
+    /// Create a context state account containing zero-knowledge proof needed
+    /// for a confidential transfer instruction.
+    pub async fn confidential_transfer_create_context_state_account<
+        S: Signer,
+        ZK: Pod + ZkProofData<U>,
+        U: Pod,
+    >(
         &self,
         context_state_account: &Pubkey,
         context_state_authority: &Pubkey,
-        ciphertext_validity_proof_data: &BatchedGroupedCiphertext3HandlesValidityProofData,
-        ciphertext_validity_proof_signer: &S,
+        proof_data: &ZK,
+        split_instruction: bool,
```

Comment:
nit: this parameter name is a little confusing -- can we call it `split_account_creation_and_proof_verification`? Or maybe you might have something better?

Reply:
Agreed. I think `split_account_creation_and_proof_verification` is also the best I can come up with, so I'll use it. Thanks!